src/host/term_textColor.c

```diff
@@ -10,6 +10,30 @@
 static int s_currentColor = -1;
 #endif
 
+#if PLATFORM_WINDOWS
+int canUseColors() {
+	return 1;
+}
+#else
+int canUseColors() {
+	return isatty(1);
+}
+#endif
+
+static const char *getenvOrFallback(const char *var, const char *fallback) {
+	const char *value = getenv(var);
+	if (value)
+		return value;
+	else
+		return fallback;
+}
+
+static int shouldUseColors() {
+	// CLICOLOR* documented at: http://bixense.com/clicolors/
+	return ((getenvOrFallback("CLICOLOR", "1")[0] != '0') && canUseColors())
+		|| getenvOrFallback("CLICOLOR_FORCE", "0")[0] != '0';
```

Comment:
I feel like, given the code below, that this function could be called a LOT...
Like, it appears to do this work every time you wanna change colour, is that right?
Can't this just be executed once at startup?

Reply:
I doubt it's of bit impact... we don't set colors that often... only for errors, which are fatal generally, so it's a one time event...

I do agree that we could at startup however.. if @lanurmi has the time to do that, that would be great? but if not, I don't want to hold off for too long and let this PR sit around.

Reply:
Yeah, give him a few days, and if not... just merge ;)

Reply:
As long as premake doesn't color each character in regular output with a different color, is there a scenario where `term_doSetTextColor()` would be called a lot in the first place? What does one need to do wrong to get hundreds or thousands of colored lines (i.e. warnings or errors) currently?

I can make it cache the result, sure, but it adds some complexity, and we don't even know if `shouldUseColors()` is a hotspot in profiling.

Reply:
> is there a scenario where `term_doSetTextColor()` would be called a lot in the first place?

The unit tests now do this, sorry! It will be called 8200+ times when running the unit tests now, so it might be a good idea to cache the result. Additionally, Premake *core* might not call it a lot in regular usage, but that doesn't mean that modules can't call it a lot for any number of reasons.

Reply:
Lets merge it, and I'll take care of the caching...